utils/babeltrace_thapi.in

```diff
@@ -287,13 +287,18 @@ def create_and_run_graph(command, inputs)
       $stdout.flush
       File.write(name, cli)
     end
-
     graph.run
-    modify_metadata(command, inputs.first)
   end
 
-  no_gc(command, inputs)
-  GC.start
+  begin
+    no_gc(command, inputs)
+  rescue ArgumentError
```

Comment:
What happens if you give an incorrect folder as an input?

Reply:
```
applenco@aurora-uan-0010:~> ~/THAPI/build/ici/bin/babeltrace_thapi -r foo/
/opt/aurora/24.086.0/spack/oneapi/0.7.0/install/2024.04.15.002/linux-sles15-x86_64/oneapi-2024.04.15.002/ruby-3.1.0-pw6tzumq2ynf6v3pwy53mgjnue6j5u5k/lib/ruby/3.1.0/psych.rb:670:in `initialize': No such file or directory @ rb_sysopen - foo/thapi_metadata.yaml (Errno::ENOENT)
	from /opt/aurora/24.086.0/spack/oneapi/0.7.0/install/2024.04.15.002/linux-sles15-x86_64/oneapi-2024.04.15.002/ruby-3.1.0-pw6tzumq2ynf6v3pwy53mgjnue6j5u5k/lib/ruby/3.1.0/psych.rb:670:in `open'
	from /opt/aurora/24.086.0/spack/oneapi/0.7.0/install/2024.04.15.002/linux-sles15-x86_64/oneapi-2024.04.15.002/ruby-3.1.0-pw6tzumq2ynf6v3pwy53mgjnue6j5u5k/lib/ruby/3.1.0/psych.rb:670:in `load_file'
	from /home/applenco/THAPI/build/ici/bin/babeltrace_thapi:260:in `thapi_metadata'
	from /home/applenco/THAPI/build/ici/bin/babeltrace_thapi:311:in `raise_if_command_invalid'
	from /home/applenco/THAPI/build/ici/bin/babeltrace_thapi:417:in `<main>'
```

Reply:
Fine as long as it still crashes. I was worried we were preventing all crashes here.